### PR TITLE
Cleanup layer CMake code

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: layersvt/CMakeLists.txt [hunk 1]
```diff
@@ -92,48 +76,31 @@ endif ()
...
-            CXX_STANDARD_REQUIRED ON
-            CXX_EXTENSIONS OFF
-        )
-        set_target_properties(copy-${target}-def-file PROPERTIES FOLDER "VkLayer_${target}")
+        add_library(VkLayer_${target} SHARED ${ARGN})
+        if (MSVC)
+            target_link_options(VkLayer_${target} PRIVATE /DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_${target}.def)
+        elseif(MINGW)
+            target_sources(VkLayer_${target} PRIVATE /DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_${target}.def)
+        endif()
```
Comment (charles-lunarg) [inline layersvt/CMakeLists.txt:84]: Weird that this is different.

Comment (juan-lunarg) [reply layersvt/CMakeLists.txt]: MINGW doesn't use the MSVC linker it uses a modified version of the GNU linker

Comment (juan-lunarg) [reply layersvt/CMakeLists.txt]: Also see updated fix what I had there was wrong

File: layersvt/CMakeLists.txt [hunk 2]
```diff
@@ -92,48 +76,31 @@ endif ()
...
-            CXX_STANDARD_REQUIRED ON
-            CXX_EXTENSIONS OFF
-        )
-        set_target_properties(copy-${target}-def-file PROPERTIES FOLDER "VkLayer_${target}")
+        add_library(VkLayer_${target} SHARED ${ARGN})
+        if (MSVC)
+            target_link_options(VkLayer_${target} PRIVATE /DEF:${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_${target}.def)
+        elseif(MINGW)
+            target_sources(VkLayer_${target} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/VkLayer_${target}.def)
+        endif()
```
Comment (juan-lunarg) [inline layersvt/CMakeLists.txt:84]: @charles-lunarg

Comment (charles-lunarg) [reply layersvt/CMakeLists.txt:84]: I see... Well, this code is necessary then.

Comment (juan-lunarg) [reply layersvt/CMakeLists.txt:84]: Yeah this is what we do in VVL:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/f046acda58667e92e7e16e006f693d220139148f/layers/CMakeLists.txt#L282

**Review by charles-lunarg — APPROVED** [review]

**Pull request merged**